libs/community/langchain_community/vectorstores/sqlserver.py

```diff
@@ -132,6 +218,66 @@ def add_texts(
         # Insert the embedded texts in the vector store table.
         return self._insert_embeddings(texts, embedded_texts, metadatas, ids)
 
+    def drop(self) -> None:
+        """Drops every table created during initialization of vector store."""
+        logging.info("Dropping vector store: %s.", self.table_name)
+        with Session(bind=self._bind) as session:
+            # Drop all the tables associated with the session bind.
+            Base.metadata.drop_all(session.get_bind())
+
+    def _search_store(
+        self, embedding: List[float], k: int, filter: Optional[dict] = None
+    ) -> List[Any]:
+        # The filter variable will be passed as an argument to the filter function
+        # filter_clause = self._create_filter_clause(filter)
+        with Session(self._bind) as session:
+            results = (
+                session.query(
+                    _embedding_store,
+                    text(
+                        """VECTOR_DISTANCE(:distancestrategy, 
+                        JSON_ARRAY_TO_VECTOR(:embedding), embeddings) AS distance"""
+                    ).bindparams(
+                        bindparam(
+                            "distancestrategy",
+                            self.distance_strategy,
+                            literal_execute=True,
+                        ),
+                        bindparam(
+                            "embedding", json.dumps(embedding), literal_execute=True
+                        ),
+                    ),
+                )
+                .filter()
+                .order_by(asc(text("distance")))
+                .limit(k)
+                .all()
+            )
+        return results
```

Comment:
What will this result look like? Can you add an example?

Reply:
This will return a list of tuple of EmbeddingStore object (which is a row in the vector store) and its score.

[(<langchain_community.vectorstores.sqlserver.SQLServer_VectorStore._get_embedding_store.<locals>.EmbeddingStore object at 0x000002B73C6F8610>, 0.5401342754176245), (<langchain_community.vectorstores.sqlserver.SQLServer_VectorStore._get_embedding_store.<locals>.EmbeddingStore object at 0x000002B73C6F8850>, 0.6400343794736375)]